### PR TITLE
refactor(apis): add `rollingUpdate`  field

### PR DESCRIPTION
**Pull request by zyy17**

### CODE REVIEW

File: controllers/greptimedbstandalone/deployer.go [hunk 1]
```diff
@@ -253,6 +253,10 @@ func (b *standaloneBuilder) BuildStatefulSet() deployer.Builder {
 			},
 			Template:             b.generatePodTemplateSpec(),
 			VolumeClaimTemplates: b.generatePVCs(),
+			UpdateStrategy: appsv1.StatefulSetUpdateStrategy{
+				Type:          appsv1.RollingUpdateStatefulSetStrategyType,
```
Comment (daviderli614) [inline controllers/greptimedbstandalone/deployer.go:257]: Can we support StatefulSet updateStrategy type of OnDelete?
https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#update-strategies

File: controllers/greptimedbcluster/deployers/meta.go [hunk 2]
```diff
@@ -247,6 +247,10 @@ func (b *metaBuilder) BuildDeployment() deployer.Builder {
 				},
 			},
 			Template: *b.generatePodTemplateSpec(),
+			Strategy: appsv1.DeploymentStrategy{
+				Type:          appsv1.RollingUpdateDeploymentStrategyType,
+				RollingUpdate: b.Cluster.Spec.Meta.RollingUpdate,
```
Comment (daviderli614) [inline controllers/greptimedbcluster/deployers/meta.go:252]: Can we support deployment update strategy type of Recreate?
https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy

File: controllers/greptimedbstandalone/deployer.go (hunk 1 above)
Comment (zyy17) [reply controllers/greptimedbstandalone/deployer.go:257]: I don't think we need `OnDelete`. It's a legacy option. We always use rolling update.

File: controllers/greptimedbcluster/deployers/meta.go (hunk 2 above)
Comment (zyy17) [reply controllers/greptimedbcluster/deployers/meta.go:252]: We don't need `Recreate`. In our scenario, `RollingUpdate` is enough.

File: apis/v1alpha1/defaulting.go [hunk 3]
```diff
@@ -427,3 +433,58 @@ func defaultReadinessProbe() *corev1.Probe {
 		FailureThreshold: 10,
 	}
 }
+
+// Same as the default rolling update strategy of Deployment.
+func defaultRollingUpdateForDeployment() *appsv1.RollingUpdateDeployment {
```
Comment (daviderli614) [inline apis/v1alpha1/defaulting.go:438]: We should not set the default RollingUpdate value here? kubernetes will set the default value for deployment and statefulset.

Comment (zyy17) [reply apis/v1alpha1/defaulting.go:438]: The default value is the same as the native K8s configuration.

**Review by daviderli614 — APPROVED** [review]

**Pull request merged**